### PR TITLE
Add Harness requirements formatting

### PR DESCRIPTION
**Pull request by cjrkoa**
Fixes issue #30 
Trying to figure out if putting the square on the same line as the requirements is possible

### CODE REVIEW

**Comment by dyc3** [comment]
Is this supposed to be a draft?

File: harness-requirements.typ [hunk 1]
```diff
@@ -11,12 +11,22 @@ Requirement priority is in ascending order, with P1 being the highest priority a
...
+	else if(pri == HIGH) { rgb("#ff390a") }
+	else if(pri == MEDIUM) { rgb("#ecff09") }
+	else if(pri == LOW) { rgb("#1ae53e") }
+	else if(pri == VERY_LOW) { rgb("#1800FF") }
 };
 
 #let req(text, necessity, pri) = {
-	priority(pri) + [ ] + necessity + [ ] + text
+	"-" + necessity + [ ] + text + [ ] + square(size: 5pt, fill: priority(pri)) + [ ] + "\n"
 };
```
Comment (dyc3) [inline harness-requirements.typ:30]: The priority number is not rendered anymore. It just shows a color.

The `priority` function should do everything required to render the priority number, colors, shapes, whatever.

**Comment by cjrkoa** [comment]
> Is this supposed to be a draft?

Yes I remember you mentioning me making a draft PR yesterday

File: harness-requirements.typ [hunk 2]
```diff
@@ -18,15 +18,40 @@ Requirement priority is in ascending order, with P1 being the highest priority a
...
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#1ae53e"), [P] + str(pri)) 
+		}
+	else if(pri == VERY_LOW) { 
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#1800FF"), [P] + str(pri)) 
+		}
```
Comment (dyc3) [inline harness-requirements.typ:50]: Let's deduplicate this a bit more.

File: harness-requirements.typ [hunk 3]
```diff
@@ -1,22 +1,57 @@
 = Test Harness Requirements
-
+//make the formatting of requirements pretty
```
Comment (dyc3) [inline harness-requirements.typ:2]: remove comment

File: harness-requirements.typ [hunk 4]
```diff
@@ -1,22 +1,57 @@
...
 == Requirements
 
-#let mustHave = [Must Have]
-#let shouldHave = [Should Have]
-#let couldHave = [Could Have]
-#let wouldBeNiceToHave = [Would Be Nice To Have]
+#let mustHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Must Have]
+#let shouldHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Should Have]
+#let couldHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Could Have]
+#let wouldBeNiceToHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Would Be Nice To Have]
```
Comment (dyc3) [inline harness-requirements.typ:12]: You should also deduplicate this a bit more. Make a function for it.

**Review by dyc3 — CHANGES_REQUESTED** [review]
Looks good visually, needs to be more DRY

File: harness-requirements.typ [hunk 5]
```diff
@@ -1,22 +1,57 @@
...
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#1ae53e"), [P] + str(pri)) 
+		}
+	else if(pri == VERY_LOW) { 
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#1800FF"), [P] + str(pri))
```
Comment (dyc3) [inline harness-requirements.typ:49]: The background and text color contrast for P5 is very low, making it hard to read.

**Review by dyc3 — APPROVED** [review]

**Pull request merged**